lightkurve/correctors/designmatrix.py

```diff
@@ -379,44 +429,396 @@ def __getitem__(self, key):
+        return self.X[:, loc].toarray()
+
+    @property
+    def shape(self):
+        """Tuple specifying the shape of the matrix as (n_rows, n_columns)."""
+        return self.X.shape
+
+    def __repr__(self):
+        return '{} SparseDesignMatrix {}'.format(self.name, self.shape)
+
+    def join(self, matrix):
+        """ Join two designmatrices, return a design matrix collection """
+        return SparseDesignMatrixCollection([self, matrix])
+
+    def to_dense(self):
+        return DesignMatrix(self.values, name=self.name,
+                            columns=self.columns, prior_mu=self.prior_mu,
+                             prior_sigma=self.prior_sigma)
+
+
+class SparseDesignMatrixCollection(DesignMatrixCollection):
+    """A set of design matrices."""
+    def __init__(self, matrices, warn=True):
+        if not np.all([issparse(m.X) for m in matrices]):
+            if warn:
+                warnings.warn(('Not all matrices are `SparseDesignMatrix` objects. '
+                                'Dense matrices will be converted to sparse matrices.'), LightkurveWarning)
+            sparse_matrices = []
+            for m in matrices:
+                if isinstance(m, DesignMatrix):
+                    sparse_matrices.append(m.copy().to_sparse())
+                else:
+                    sparse_matrices.append(m)
+            self.matrices = sparse_matrices
+        else:
+            self.matrices = matrices
+        self.X = hstack([m.X for m in self.matrices], format='csr')
+        self._child_class = SparseDesignMatrix
+        self.validate()
+
+    def plot(self, ax=None, **kwargs):
+        """Visualize the design matrix values as an image.
+
+        Uses Matplotlib's `~lightkurve.utils.plot_image` to visualize the
+        matrix values.
+
+        Parameters
+        ----------
+        ax : `~matplotlib.axes.Axes`
+            A matplotlib axes object to plot into. If no axes is provided,
+            a new one will be created.
+        **kwargs : dict
+            Extra parameters to be passed to `.plot_image`.
+
+        Returns
+        -------
+        `~matplotlib.axes.Axes`
+            The matplotlib axes object.
+        """
+        temp_dm = SparseDesignMatrix(hstack([d.X for d in self]))
+        ax = temp_dm.plot(**kwargs)
+        ax.set_title("Design Matrix Collection")
+        return ax
+
+    def __repr__(self):
+        return 'SparseDesignMatrixCollection:\n' + \
+                    ''.join(['\t{}\n'.format(i.__repr__()) for i in self])
+
+
+
 ####################################################
 # Functions to create commonly-used design matrices.
 ####################################################
 
-def create_spline_matrix(x, n_knots=20, degree=3, name='spline',
-                         include_intercept=False):
-    """Returns a `.DesignMatrix` which models splines using `patsy.dmatrix`.
 
+@jit(nopython=True)
+def basis(x, degree, i, knots):
```

Comment:
Would this be a better name?

```suggestion
def spline_basis_vector(x, degree, i, knots):
```

Reply:
Sure! This is actually a hidden function I'd rather not have the user know about (it's hard to use) so I think I'll call it `_spline_basis_vector`